tests/suites/test_suite_pk.function

```diff
@@ -2008,9 +2012,7 @@ void pk_sign_ext(int pk_type, int curve_or_keybits, int key_pk_type, int md_alg)
     mbedtls_pk_init(&pk);
     MD_OR_USE_PSA_INIT();
 
-    TEST_EQUAL(mbedtls_pk_setup(&pk,
-                                mbedtls_pk_info_from_type(pk_type)), 0);
-    TEST_EQUAL(pk_genkey(&pk, curve_or_keybits), 0);
+    TEST_EQUAL(pk_genkey(&pk, pk_type, curve_or_keybits), 0);
 
```

Comment:
In `pk_psa_wrap_sign_ext` below, there's a call to `mbedtls_rsa_gen_key`. We should replace this one since it's the same pattern that was used in `pk_genkey`, just done directly rather than through a function that also works for EC keys.